csharp/ql/src/semmle/code/csharp/dataflow/internal/rangeanalysis/SignAnalysisSpecific.qll

```diff
@@ -75,6 +75,12 @@ predicate unknownIntegerAccess(Expr e) {
   or
   //method call, local function call, ctor call, ...
   e instanceof CS::Call and e.getType() instanceof NumericOrCharType
+  or
+  // checked
+  e instanceof CS::CheckedExpr and e.getType() instanceof NumericOrCharType
+  or
+  // unchecked
+  e instanceof CS::UncheckedExpr and e.getType() instanceof NumericOrCharType
```

Comment:
Why not defined in `specificSubExprSign`, i.e. the signs of an (un)checked expression are the signs of the underlying expression.

Reply:
When we have `checked` or `unchecked`, we are working on some code that can overflow. Sign and range analysis doesn't handle overflow, so I think when we see these constructs we can be super conservative and assume we don't know anything about the sign. What do you think?

Reply:
Good point.